### PR TITLE
Adding county information to API

### PR DESCRIPTION
**Pull request by JKSenthil**
County information is retrieved from CSBS (https://www.csbs.org/information-covid-19-coronavirus). Endpoint is `/v2/locations?source=csbs`. Added unit tests and added documentation in readme for the county addition.

### CODE REVIEW

**Comment by JKSenthil** [comment]
Yeah, having a middleware would definitely be better. Feel free to let me know if you have any questions or other comments!

**Comment by ExpDev07** [comment]
I started some ground-work here: https://github.com/ExpDev07/coronavirus-tracker-api/pull/115.

**Comment by ExpDev07** [comment]
Hello, @JKSenthil! It should now be done, if you still want to add your source, a couple of things:

You're only allowed to return instances of ``Location``. You may create a subclass like (suggested implementation):

```python
from . import Location

class CbsLocation(Location):
    """
    A CBS location.
    """

    def __init__(self, id, country, state, county, coordinates, confirmed, deaths):
        super().__init__(
            # General info.
            id, 'US', state, coordinates,

            # Statistics.
            confirmed=confirmed,
            deaths=deaths,
            recovered=0, # not provided
        )

        # State and county.
        self.county = county
        self.state = state

    def serialize(self):
        """
        Serializes the location into a dict.


        :returns: The serialized location.
        :rtype: dict
        """
        serialized = super().serialize()

        # Update with new fields.
        serialized.update({
            'state'   : self.state,
            'county': self.county,
        })

        # Return the serialized location.
        return serialized
```

Then simply create your service for retrieving the locations, and register it in ``data/__init__.py``. Now your work is done. You don't have to do anything in the routes. You'll have fields such as "province" and "state" saying the same, but that's not really a big deal, and will provide compatibility across the data-sources.

I can work on implementing a more universal form for sending query params through to the services so we don't mess up the routes later. I just did it the easy way with ``country_code``, but it should really be moved out of the routes module.

Happy coding! I might write a guide on adding data-sources later, but this ^ is a quick summary of it.

**Comment by JKSenthil** [comment]
Great, thanks for the details! I'll get to work.

File: app/services/location/csbs.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    for i, item in enumerate(data):
+        state = item['State Name']
+        county = item['County Name']
+        if county == "Unassigned" or county == "Unknown":
+            continue
+
+        confirmed = int(item['Confirmed'] or 0)
+        death = int(item['Death'] or 0)
+        coordinates = Coordinates(float(item['Latitude']), float(item['Longitude']))
+        last_update = item['Last Update']
```
Comment (ExpDev07) [inline app/services/location/csbs.py]: Should be parsed as ISO with an added 'Z' at the end. Use ``import datetime from datetime`` to achieve this.

**Comment by JKSenthil** [comment]
I have finished up the csbs service and integrated it via the middleware, so everything is working. I think the only "issue" is that is `/v2/locations.py`, the serialising step passes in a `timelines` which the csbs doesn't use. For now, I pass in a dummy timelines in serialize(), is that fine?

**Comment by ExpDev07** [comment]
@JKSenthil 
> the serialising step passes in a timelines which the csbs doesn't use. For now, I pass in a dummy timelines in serialize(), is that fine?

Yes, that's fine. I'll fix it later. It's really just about checking if the ``Location`` is an instanceof ``TimelinedLocation`` before we pass in the timelines argument in the routes, I'll sort it, don't worry, for now it's fine.

File: app/services/location/csbs.py (hunk 1 above)
Comment (JKSenthil) [reply app/services/location/csbs.py]: fixed!

**Comment by ExpDev07** [comment]
Hey @all-contributors please add @JKSenthil for code, documentation, and tests.

**Pull request merged**